### PR TITLE
Cardtypes

### PR DESCRIPTION
**Pull request by bbqbaron**
The high-level change here is "let `Card` derive what variant it's gotten". I don't have a strong opinion about what to do here, but this change is based on some assumptions:
* JSON shouldn't have to carry artifacts of our type system, ie variants are just useful TS tags
* if we can infer behavior from data we already have, that seems reasonable until the complexity gets out of control
* I can make TS shut up about this without, like, walking the entire JSON tree adding `variant` with Ramda functions that TS doesn't understand...which I definitely didn't try or anything 😉 

So basically this uses the presence of eg `scene` to derive the variant rather than stamping it on props. Thoughts? Again, not asserting this is The Only Way (TM), especially since I haven't worked in typed systems in a bit.

### CODE REVIEW

**Comment by bbqbaron** [comment]
This does seem less elegant from a type-system perspective, since it leaves some of the decisions to runtime by being all duck-typey 🤔. I'm not sure whether to be a perfectionist about it.

**Comment by bbqbaron** [comment]
I guess the props being a sum of the card types means that it would be reeeeeasonably hard to fuck up those functions that use the `is` checks.
